noxfile.py

```diff
@@ -180,3 +180,27 @@ def make(session: nox.Session):
         *(args.make_args or ("clean", "coredocs")),
     ]
     session.run("make", "-C", "docs/docsite", *make_args, external=True)
+
+
+@nox.session
+def tag(session: nox.Session):
+    """
+    Determine the missinlg ansible-core releases,
+    create corresponding tags for each release in the ansible-documentation repo,
+    and then push them
+    """
+    install(session, req="tag")
+
+    DEFAULT_REPO = "https://github.com/ansible/ansible"
+
+    args = list(session.posargs)
+    if not any(arg.startswith("--core") for arg in args):
+        tmpdir = session.create_tmp()
+        session.run("git", "clone", "--quiet", DEFAULT_REPO, tmpdir, external=True)
+        args.extend(["--core", tmpdir])
+
+    # If run without any commands, default to "tag"
+    if not any(arg.startswith(("hash", "mantag", "new-tags", "tag")) for arg in args):
+        args.append("tag")
```

Comment:
It might be nice to first check for new tags. I'd kind of like to have a session where you do `nox -s tag` then it:

- Checks if the core checkout exists.
- Compares the core checkout against the upstream HEAD to make sure it's up to date.
- Runs `new-tags` to fetch any new tags.
- Runs `tag` to create new tags.

Reply:
I can't find the comment, but I thought that when I was working on the tagging documentation someone told me that running `tag` implicitly does the check that `new-tags` would do, and that most everything other than the `tag` command for `tag.py` was for debugging.

That being said, the rest of the notes here sound like a better approach to me, and I'll be wrapping that into my next pass.

Reply:
@x1101 Yes that was probably @gotmax23 (who is pretty amazing). I can see in the tagger that it does look like that is what is happening: https://github.com/ansible/ansible-documentation/blob/d4c6186c2957ec4c061470177346debf7ced0403/hacking/tagger/tag.py#L413

So, yeah, please ignore me on that one.